Core/Src/ACDC_SPI.c

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file ACDC_SPI.h
+ * @author Liam Bramley
+ * @brief Implementation of SPI driver
+ * 
+ * This file includes functions to initalize the SPI, transfer and recieve data.
+ * 
+ * @version 0.1
+ * @date 2024-01-30
+ * @copyright Copyright (c) 2023-2024
+ * 
+ */
+
+#include "ACDC_SPI.h"
+
+
+/*
+{
+  __IO uint32_t CR1;      /* SPI control register 1         */
+  __IO uint32_t CR2;      /* SPI control register 2         */
+  __IO uint32_t SR;       /* SPI status register            */
+  __IO uint32_t DR;       /* SPI data register              */
+  __IO uint32_t CRCPR;    /* SPI CRC polynomial register    */
+  __IO uint32_t RXCRCR;   /* SPI RX CRC register            */
+  __IO uint32_t TXCRCR;   /* SPI TX CRC register            */
+  __IO uint32_t I2SCFGR;  /* SPI_I2S configuration register 
+ SPI_TypeDef;
+*/
+
+void SPI_Initalize(const SPI_TypeDef *SPI) {
+    if(SPI == SPI1){
+        RCC-> APB2ENR |= RCC_APB2ENR_IOPBEN;
```

Comment:
You should not need this, GPIO_PinDirection takes care of enabling the IO Clock Registers